scripts/test_commit_range.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/sh
+###########################################################################
+#
+#  Copyright 2016-2018, ARM Limited, All Rights Reserved
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+###########################################################################
+
+# Check if all commits since a specified commit (not including that commit)
+# build and pass some basic checks.
+
```

Comment:
I chose not to use `set -e` so that the script can return to the original branch after a test fails. Probably we want `set -e` for all areas except the actual commit testing function.

Reply:
Use `set -e` and allow the commit testing function to fail. E.g.

```
set -e
…
if ! test_this_commit; then
  echo >&2 $commit failed
  status=1
fi
```